packages/notifier/README.md

```diff
@@ -1,6 +1,263 @@
+  updateState: val => console.log('non-final', val),
+  finish: completion => console.log('finished', completion),
+  fail: reason => console.log('failed', reason),
+});
+observeIteration(subscription, observer);
+// eventually prints
+// non-final a
+// non-final b
+// finished done
+```
+
+The difference between the two pairs is that the NotifierKit is a lossy
+conveyor of non-final values whereas the SubscriptionKit is a lossless conveyor
+of non-final values. They both losslessly convey termination, so the
+SubscriptionKit losslessly conveys the entire async iteration.
+
+Had the example above started with
+```js
+const { updater, notifier } = makeNotifierKit();
+```
+propagating the variable renaming through the rest of the code, the code above
+remains correct, but Alice and Bob may each have missed either or both of the
+non-final values.
+
+
+## Distributed Operation
+
+Both pairs can be used in a
+multicast manner with good distributed systems properties, where there is only
+one producing site but any number of consuming sites. The producer is not
+vulnerable to the consumers and the consumers are not vulnerable to each other.
+For distributed operation, all the iteration values---non-final values,
+successful completion value, failure reason---must be `Passable`, i.e., must be
+values that can somehow be passed betwen vats. The rest of this page assumes all
+these values are Passable.
+
+The maker is used to make the pair on the producer's site. As a result, both
+the iterationObserver and the initial asyncIterable are on the producer's site.
+If Paula the producer sends Bob the asyncIterable, Bob receives a possibly
+remote reference to the async iterable Bob's code above is still correct if he
+uses this
+reference directly, since `observeIteration` only needs its first argument to
+be an `ERef<AsyncIterable<Passable>>`, i.e., a reference of some sort to an
+AsyncIterable conveying Passable values. This reference may be a local
+AsyncIterable, a remote presence of an AsyncIterable, or a local or remote
+promise for an AsyncIterable. `observeIteration` only sends it eventual
+messages using `E` (equivalent to the tildot syntax `~.`), and so doesn't care
+about these differences.
+
+Although Bob's code is correct, it is sub-optimal. Its distributed systems
+properties are not terrible, but Bob can do better using the
+`getSharableSubsciptionInternals()` method provided by both NotifierKit and
+SubscriptionKit. This enables consumer Bob to make a local AsyncIterable that
+coordinates better with producer Paula's IterationObserver. Alice's code is
+less forgiving. She's using JavaScript's `for-await-of` loop which requires a
+local AsyncIterable. It cannot handle a remote reference to an AsyncIterable
+at
+Paula's site. Alice has no choice but to make an AsyncIterable at her site.
+Using `getSharableSubsciptionInternals()` is the best way for her to do so. She
+can replace her last line, the call to `consume(subscription)` with
```

Comment:
I am wondering whether there might be a way to hide this concern from users. This is a lot to chew.

These docs might benefit from being overall driven by a sequence of motivating use cases with concrete domains, like the current time, progress, ETA, absolute values vs deltas.

Reply:
> I am wondering whether there might be a way to hide this concern from users. This is a lot to chew.

There is --- building more direct support for Unum replication into `marshal` and the comm systems built on it. However, it requires some kind of name-based code registry be privileged. This also comes up for the ERTP MathHelpers. We don't yet have a worked out design for reconciling such a name-based registry with permissionless-ness, so we have postponed the issue. But we know we'll need this.

Reply:
> These docs might benefit from being overall driven by a sequence of motivating use cases with concrete domains, like the current time, progress, ETA, absolute values vs deltas.

They would indeed, but please not in this PR. For now I'm trying only for precision and a certain basic level of readability. Does it seem adequate to merge in its current state?